ui/apps/platform/src/Containers/NetworkGraph/components/DefaultFakeGroup.tsx

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable no-void */
+/* eslint-disable no-cond-assign */
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
+/* eslint-disable no-return-assign */
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Topology/topology-components';
+import {
+    Layer,
+    LabelPosition,
+    LabelBadge,
+    Ellipse,
+    NodeLabel,
+    createSvgIdUrl,
+    useCombineRefs,
+    useHover,
+    useSize,
+    useDragNode,
+} from '@patternfly/react-topology';
+
+const DefaultFakeGroup = ({
+    className,
+    children,
+    element,
+    selected,
+    onSelect,
+    hover,
+    label,
+    secondaryLabel,
+    showLabel = true,
+    truncateLength,
+    collapsedWidth,
+    collapsedHeight,
+    getCollapsedShape,
+    collapsedShadowOffset = 8,
+    dragNodeRef,
+    dragging,
+    labelPosition,
+    badge,
+    badgeColor,
+    badgeTextColor,
+    badgeBorderColor,
+    badgeClassName,
+    badgeLocation,
+    labelIconClass,
+    labelIcon,
+    labelIconPadding,
+}) => {
+    // eslint-disable-next-line @typescript-eslint/naming-convention, no-underscore-dangle
+    let _a: number | undefined;
+    const [hovered, hoverRef] = useHover();
+    const [labelHover, labelHoverRef] = useHover();
+    const dragLabelRef = useDragNode()[1];
+    const [shapeSize, shapeRef] = useSize([collapsedWidth, collapsedHeight]);
+    const refs = useCombineRefs(hoverRef, dragNodeRef, shapeRef);
+    const isHover = hover !== undefined ? hover : hovered;
+    const childCount: number = element.data.flows;
+    const [badgeSize, badgeRef] = useSize([childCount]);
+    const groupClassName = css(
+        styles.topologyGroup,
+        className,
+        dragging && 'pf-m-dragging',
+        selected && 'pf-m-selected'
+    );
+    const ShapeComponent = getCollapsedShape ? getCollapsedShape(element) : Ellipse;
+    const filter = isHover || dragging ? createSvgIdUrl('NodeShadowsFilterId--hover') : undefined;
+    return React.createElement(
+        'g',
+        { ref: labelHoverRef, onClick: onSelect, className: groupClassName },
+        React.createElement(
+            Layer,
+            { id: 'groups' },
```

Comment:
there's some stuff here that causes typescript errors that i'm not sure how to resolve -- should i just make this file `js` for now? the file i copied a lot of this from in the `react-topology` library is also a js file fwiw

Reply:
Reach out to @pedrottimark before we punt back to `.js`. He might have some TS insight for this.